issuetracker/src/main/java/org/openjdk/skara/issuetracker/Issue.java

```diff
@@ -188,6 +188,50 @@ default URI nonTransformedWebUrl() {
 
     List<Link> links();
 
+    /**
+     * Get the links which have the relationships listed in the arguments.
+     * @param relationships the link relationship
+     * @return the related links
+     */
+    default List<Link> linksWithRelationships(List<String> relationships) {
+        var result = new ArrayList<Link>();
+        if (relationships == null) {
+            return result;
+        }
+        for (var link : links()) {
+            if (link.relationship().isPresent() && relationships.contains(link.relationship().get())) {
+                result.add(link);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get the corresponding csr link of current issue if it exists.
+     * @return the csr link
+     */
+    default Optional<Link> csrLink() {
+        var links = linksWithRelationships(List.of("csr for"));
+        if (links != null && !links.isEmpty()) {
```

Comment:
From what I can tell, `links` can never be null here. 

I think this method could be expressed a bit more elegantly with stream usage, like this:
```
return links.stream().findAny();
```